### ISSUE TITLE
ARM assembly syntax changed in LLVM for arm-64-osx

### ISSUE THREAD

**Issue by dsharletg** [halide/Halide#5786]
This is breaking simd_op_check, it's not clear how we should fix this.

**Comment by steven-johnson** [comment]
Seems odd that it would be different for arm-64-osx vs arm-64-linux, but I'll take a look

**Comment by dsharletg** [comment]
I misdiagnosed this, this is just because there's a new osx ARM build bot, which I didn't realize. I think Andrew is looking at this already.

**Comment by abadams** [comment]
Yeah this is a difference in assembly printing for M1 somehow. The big issue is that the registers are not type-annotated, but the output type for the instruction *is*. These types are not the same, so there's no obvious way to continue type-checking without adding a third expected format for each instruction.

**Comment by zvookin** [comment]
Is there a flag to request different assembly formatting?

-Z-
